### ISSUE TITLE
Unable to update device after deleting its device redundancy group if the device DRG priority is set

### ISSUE THREAD

**Issue by nrnvgh** [nautobot/nautobot#4718]
### Environment
* Nautobot version (Docker tag too if applicable): 1.6.4
* Python version: 3.11.6
* Database platform, version: mysql 8.1.0
* Middleware(s): pynautobot, GUI, Swagger UI

### Steps to Reproduce
1. Create a device redundancy group (DRG)
2. Create a device
3. Assign device to redundancy group with a priority
4. Delete device redundancy group
5. Attempt to make an edit to the device

Note that with the DRG deleted, it's no longer listed in the device edit screen but the DRG priority remains set; since having a DRG priority set without having a corresponding DRG set fails validation, no edits can be made to the device until either the priority is removed or a new DRG is configured. Attempts to edit the device result in the error see in Observed Behavior, below. Bonus: If you attempt to add such a device to a Virtual Chassis, you get this error:

![Screenshot 2023-10-26 at 5 32 26 PM](https://github.com/nautobot/nautobot/assets/120611026/c911bc78-d9ba-42e6-b893-3dbceb9836ff)

### Expected Behavior
I can absolutely see that changing this behavior may be more trouble than it's worth, however in a perfect world it feels to me as though either any devices using the deleted redundancy group should have their priority set to `None` or DRG deletion should fail until/unless no devices are using it. That said, I don't know the code base well enough to know which (or if either) of those solutions is correct. 

### Observed Behavior
After the DRG was deleted and I was unable to make any updates to the device without first nulling out the device DRG priority. returned message is:

```json
{
  "device_redundancy_group_priority": [
    "Must assign a redundancy group when defining a redundancy group priority."
  ]
}
```

Thoughts?

**Comment by nrnvgh** [comment]
PR created with a patch following the style of VirtualChassis; at first glance, it appears the same fix may be needed in `develop`, but I haven't tested that myself. If it is needed, let me know how you'd like to proceed, workflow-wise.

**Comment by HanlinMiao** [comment]
Hi @nrnvgh, thank you for the issue and PR. The fix looks great, but do you have the bandwidth to add a unit test to test this bug fix?

**Comment by nrnvgh** [comment]
I spent some time today working up a test case. I have one which works, but in the process of developing it I tripped over something else, closely related. If I have a device with a redundancy group and a priority assigned  and then do this in a test case:

```python
self.device.device_redundancy_group = None
self.device.save()
```

...there's no error, whereas in the GUI/API, I would get this error:

`Must assign a redundancy group when defining a redundancy group priority.`

This feels like it could be a bug related to this one, but I don't know enough about the philosophy of your codebase to know for certain. Should this throw a ValidationException or do you want to allow Device objects to fiddle with those fields? And if it is something which should be fixed, should it be rolled into this patch or should it be a new one?

**Comment by smk4664** [comment]
When doing anything through the ORM, you need to call `.validated_save()` instead of `.save()` to enforce validation.

**Comment by nrnvgh** [comment]
Thanks; I've added a test case. If acceptable, there's still an open question as to how to get this change into the `develop` branch. What's your preferred workflow for ensuring a fix is included in both the `LTM` and `develop` branches?

**Comment by glennmatthews** [comment]
General preference would be to do a PR into `develop` first, then a separate PR into `1.6` for the backport, but the exact order of operations here isn't critical so long as we're not making fixes in 1.6 and forgetting to also include them in `develop`.

**Comment by nrnvgh** [comment]
Added test case to `1.6` patch and submitted PR for patch+test case for `develop`.

**Comment by nrnvgh** [comment]
Possibly I could have handled my PR's differently; in my mind, the closing of this issue requires both PR #4737 and #4721 (for the `ltm-1.6` branch) to be applied. Thoughts?

**Comment by glennmatthews** [comment]
It got auto-closed when the PR was merged to develop; agreed that we want the fix in 1.6 as well.
